File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        // leetcode
        // k = 3, lee, eet, etc ...
        Set<Character> vowels = Set.of('a', 'e', 'i', 'o', 'u');
        
        int result = 0;
        for (int i=0; i<k; ++i) {
            if (vowels.contains(s.charAt(i))) {
                result++;
            }
        }
        int max = result;
        for (int i=k; i<s.length(); ++i) {
            if (vowels.contains(s.charAt(i))) {
                result++;
            } 
            if (vowels.contains(s.charAt(i-k))) {
                result--;
            }
            max = Math.max(max, result);
        }

        return max;
    }
}

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = {'a', 'e', 'i', 'o', 'u'}
        count = 0
        for i in range(k):
            if s[i] in vowels:
                count += 1
        # max 변수에 count 값 담아두고 count 재사용
        result = count

        for i in range(k, len(s)):
            if s[i] in vowels:
                count += 1
            if s[i-k] in vowels:
                count -= 1
            result = max(result, count)
        return result
        
        


            
        